drivers/bluetooth/hci/hci_nxp_setup.c

```diff
@@ -0,0 +1,1178 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <stddef.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/arch/cpu.h>
+
+#include <zephyr/init.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/byteorder.h>
+#include <string.h>
+
+#define LOG_LEVEL CONFIG_BT_HCI_DRIVER_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(bt_nxp_ctlr);
+
+#include "common/bt_str.h"
+
+#include "bt_nxp_ctlr_fw.h"
+
+static const struct device *uart_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_bt_uart));
+
+#define HCI_UART_CHILD_PREFIX(x) x
+#define HCI_UART_CHILD           DT_FOREACH_CHILD(DT_CHOSEN(zephyr_bt_uart), HCI_UART_CHILD_PREFIX)
+#if DT_NODE_HAS_PROP(HCI_UART_CHILD, sdio_reset_gpios)
+struct gpio_dt_spec sdio_reset = GPIO_DT_SPEC_GET(HCI_UART_CHILD, sdio_reset_gpios);
+#endif /* DT_NODE_HAS_PROP(HCI_UART_CHILD, sdio_reset_gpios) */
+#if DT_NODE_HAS_PROP(HCI_UART_CHILD, w_disable_gpios)
+struct gpio_dt_spec w_disable = GPIO_DT_SPEC_GET(HCI_UART_CHILD, w_disable_gpios);
+#endif /* DT_NODE_HAS_PROP(HCI_UART_CHILD, w_disable_gpios) */
+
+struct nxp_ctlr_dev_data {
+	uint32_t primary_speed;
+	bool primary_flowcontrol;
+	uint32_t secondary_speed;
+	bool secondary_flowcontrol;
+};
+
+static struct nxp_ctlr_dev_data uart_dev_data;
+
+#define DI         0x07U
+#define POLYNOMIAL 0x04c11db7UL
+
+#define CRC32_LEN 4
+
+static unsigned char crc8_table[256U]; /* 8-bit table */
+static unsigned long crc_table[256U];
+static bool made_table;
+
+static void fw_upload_initCrc8(void)
```

Comment:
Avoid using camelCasing, also can't you use the CRC functions from `zephyr/sys/crc.h`?

Reply:
Fixed.

CRC8 could be replaced by zephyr/sys/crc.h. 
But CRC32 can not be replaced.